lib/util/http/github.ts

```diff
@@ -172,6 +174,76 @@ function constructAcceptString(input?: any): string {
   return acceptStrings.join(', ');
 }
 
+const MAX_GRAPHQL_PAGE_SIZE = 100;
+
+function getGraphqlPageSize(
+  fieldName: string,
+  defaultPageSize = MAX_GRAPHQL_PAGE_SIZE
+): number {
+  const cache = getCache();
+  cache.platform ??= {};
+  cache.platform.github ??= {};
+  cache.platform.github.graphqlPageCache ??= {};
+  const cachedRecord = cache.platform.github.graphqlPageCache[fieldName];
+
+  if (cachedRecord) {
+    logger.debug(
+      { fieldName, ...cachedRecord },
+      'GraphQL page size: found cached value'
+    );
+
+    const oldPageSize = cachedRecord.pageSize;
+
+    const now = DateTime.local();
+    const then = DateTime.fromISO(cachedRecord.pageLastResizedAt);
+    const expiry = then.plus({ hours: 24 });
+    if (now > expiry) {
+      const newPageSize = Math.min(oldPageSize * 2, MAX_GRAPHQL_PAGE_SIZE);
+      if (newPageSize < MAX_GRAPHQL_PAGE_SIZE) {
```

Comment:
What about the else condition here? e.g. newPageSize is exactly equal to max size

Reply:
Good point, I'm working on this now

Reply:
is this resolved?

Reply:
Current implementation doubles page size every 24 hours, and removes cache record completely once it reaches 100 or more

Reply:
If this is what we want, this one is resolved then